app/src/main/java/ca/bc/gov/bchealth/ui/healthrecord/add/FetchTestRecordFragment.kt

```diff
@@ -0,0 +1,242 @@
+    private val viewModel: FetchTestRecordsViewModel by viewModels()
+    private lateinit var savedStateHandle: SavedStateHandle
+    private val recentPhnDobViewModel: RecentPhnDobViewModel by viewModels()
+
+    companion object {
+        private const val TAG = "FetchTestRecordFragment"
+        const val TEST_RECORD_ADDED_SUCCESS = "TEST_RECORD_ADDED_SUCCESS"
+    }
+
+    override fun onViewCreated(view: View, savedInstanceState: Bundle?) {
+        super.onViewCreated(view, savedInstanceState)
+        savedStateHandle = findNavController().previousBackStackEntry!!.savedStateHandle
+        savedStateHandle.set(TEST_RECORD_ADDED_SUCCESS, -1L)
+
+        setupToolBar()
+
+        setUpPhnUI()
+
+        setUpDobUI()
+
+        setUpDotUI()
+
+        initClickListeners()
+
+        observeCovidTestResult()
+    }
+
+    private fun showLoader(value: Boolean) {
+        binding.btnSubmit.isEnabled = !value
+        binding.progressBar.isVisible = value
+    }
+
+    private fun observeCovidTestResult() {
+        viewLifecycleOwner.lifecycleScope.launch {
+            lifecycle.repeatOnLifecycle(Lifecycle.State.STARTED) {
+                viewModel.uiState.collect { state ->
+
+                    showLoader(state.onLoading)
+
+                    if (state.errorData != null) {
+                        requireContext().showError(
+                            getString(state.errorData.title),
+                            getString(state.errorData.message)
+                        )
+                    }
+
+                    if (state.queItTokenUpdated) {
+                        fetchTestRecord()
+                    }
+
+                    if (state.onTestResultFetched > 0) {
+                        savedStateHandle.set(TEST_RECORD_ADDED_SUCCESS, state.onTestResultFetched)
+                        if (binding.checkboxRemember.isChecked) {
+                            val phn = binding.edPhn.text.toString()
+                            val dob = binding.edtDob.text.toString()
+                            recentPhnDobViewModel.setRecentPhnDobData(phn, dob)
+                        }
+                        findNavController().popBackStack()
+                    }
+
+                    if (state.onMustBeQueued && state.queItUrl != null) {
+                        queUser(state.queItUrl)
+                    }
+                }
+            }
+        }
+    }
+
+    private fun initClickListeners() {
+        binding.btnSubmit.setOnClickListener {
+
+            fetchTestRecord()
+        }
+
+        binding.btnCancel.setOnClickListener {
+            findNavController().popBackStack()
+        }
+    }
+
+    private fun fetchTestRecord() {
+        val phn = binding.edPhn.text.toString()
+        val dob = binding.edtDob.text.toString()
+        val dot = binding.edtDoc.text.toString()
+
+        if (this.validatePhnNumber(
+                binding.edPhnNumber,
+                getString(R.string.phn_should_be_10_digit)
```

Comment:
This param is ignored I think. I would just embed the errors in the call itself. It makes sense that `validatePhnNumber` would know which error message to throw.

Also, the 'n' stands for number so it should just be `validatePhn`

Reply:
This will be taken care in gradual refactoring we will be doing